packages/pn-personagiuridica-webapp/src/component/Deleghe/DelegationsElements.tsx

```diff
@@ -45,10 +59,80 @@ export const Menu: React.FC<{
     setAnchorEl(event.currentTarget);
   };
 
+  const handleConfirmClick = () => {
+    if (menuType === 'delegates') {
+      void dispatch(revokeDelegation(id))
+        .unwrap()
+        .then(async () => {
+          dispatch(
+            appStateActions.addSuccess({
+              title: '',
+              message: t('deleghe.revoke-successfully'),
+            })
+          );
+        });
+    } else {
+      void dispatch(rejectDelegation(id))
+        .unwrap()
+        .then(async () => {
+          dispatch(
+            appStateActions.addSuccess({
+              title: '',
+              message: t('deleghe.reject-successfully'),
+            })
+          );
+          await dispatch(getSidemenuInformation());
+        });
+    }
+    onCloseModal();
+  };
+  const handleConfirmationError = useCallback((responseError: AppResponse) => {
+    if (Array.isArray(responseError.errors)) {
+      const managedErrors = (
+        Object.keys(ServerResponseErrorCode) as Array<keyof typeof ServerResponseErrorCode>
+      ).map((key) => ServerResponseErrorCode[key]);
+      const error = responseError.errors[0];
+      if (!managedErrors.includes(error.code as ServerResponseErrorCode)) {
+        // mettere condizione per il quale richiamo la funzione desiderata in base al menutype
+        if (menuType === 'delegates') {
```

Comment:
I should create a constant `const message = menuType === 'delegates' ? t('deleghe.revoke-error') : t('deleghe.reject-error')` and after `dispatch(appStateActions.addError({ title: '', message }));`

Reply:
done